### PR TITLE
feat(hot-toast): add an option to pass data with component

### PR DESCRIPTION
**Pull request by shhdharmen**
#54

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #54 

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/app/sections/example/example.component.ts [hunk 1]
```diff
@@ -511,3 +556,10 @@ export class DummyComponent {}
 export class InjectorComponent {
   constructor(@Inject('MESSAGE') public message: string) {}
 }
+@Component({
+  selector: 'app-data',
+  template: '{{ toastRef.data.fact }}',
+})
+export class DataComponent {
+  constructor(@Inject(HotToastRef) public toastRef: HotToastRef) {}
```
Comment (NetanelBasal) [inline src/app/sections/example/example.component.ts:564]: Can we provide a generic for data? `HotToastRef<DataType>`

File: projects/ngneat/hot-toast/src/lib/components/hot-toast/hot-toast.component.ts [hunk 2]
```diff
@@ -34,9 +37,23 @@ export class HotToastComponent implements OnInit, AfterViewInit, OnDestroy {
 
   isManualClose = false;
   context: Record<string, any>;
+  toastComponentInjector: Injector;
+
+  constructor(private injector: Injector) {}
 
   ngOnInit() {
     this.context = { ...this.toast.context, $implicit: this.toastRef };
```
Comment (NetanelBasal) [inline projects/ngneat/hot-toast/src/lib/components/hot-toast/hot-toast.component.ts:45]: Maybe it's good to be aligned with this functionality when using a template. We can let them pass `data` when using a template, and put it under the data key:

 `this.context = { data: this.toast.data, $implicit: this.toastRef };`

There is no need to create the `context` if we're not using a template.

Comment (shhdharmen) [reply projects/ngneat/hot-toast/src/lib/components/hot-toast/hot-toast.component.ts:45]: If we remove `context`, it would be a breaking change, right?

Comment (NetanelBasal) [reply projects/ngneat/hot-toast/src/lib/components/hot-toast/hot-toast.component.ts:45]: LGTM

**Review by NetanelBasal — APPROVED** [review]

**Pull request merged**